### DISCUSSION TITLE
Is there a automated/best way to filter and get digital only stuff?

### DISCUSSION THREAD

**Discussion by zealvix** [unexpectedpanda/retool#325] [General]
1. Is there a automated/best way to filter and get a list of digital only stuff for a system using this tool, or perhaps other tools?

Example: only return PSN exclusive stuff (e.g Legend of Heroes - Trails in the Sky - Second Chapter (USA) (PSN)) from "Unofficial - Sony - PlayStation Portable (PSN) (Decrypted) " DAT in nointro 's site that is not inside the redump set for English release?

Likewise for other systems where digital-only releases started, especially for some languages.

2. Does retool works with updates/dlcs(digital)?
I was using it on some dlc/updates DAT in no intro (e.g Unofficial - Nintendo - Nintendo 3DS (Digital) (Updates and DLC) ) and it seems to be returning updates from multiple language for the same game.

**Comment by unexpectedpanda** [comment]
> Example: only return PSN exclusive stuff (e.g Legend of Heroes - Trails in the Sky - Second Chapter (USA) (PSN)) from "Unofficial - Sony - PlayStation Portable (PSN) (Decrypted) " DAT in nointro 's site that is not inside the redump set for English release?

It sounds like you're asking if there's a way to compare DAT files, and only take the unique titles from one. Retool doesn't support this.

As an alternative you could make your own post filter list and share it around.

> Does retool works with updates/dlcs(digital)?

It works, and I can see 970 clones are found -- however CDN-style DAT files haven't been paid a lot of attention. At a cursory glance the main issue I see is it looks like the `(Embedded Update)` and `(Update)` tags aren't being recognized as related.

As per a Reddit thread I just found:

> The Embedded Update is the base game and the update.

I can change things so Retool only takes the latest version regardless of whether it's an embedded or standalone update. I'm not sure whether that's the best choice though, and need to do a little research.

**Comment by zealvix** [comment]
Thanks for the confirmation, looks like googling for digital only titles and manually searching is still the way to go for getting digital exclusive stuff.

For the dlc/updates, maybe its just the clone list being less updated I guess compared to the main ones. I remember quite a number of duplicates from stuff like fire emblem and Zelda (which were more apparent to me as I was checking specifically for them), though it was some months back.
If a game does not has updates/dlc for the English/USA version (maybe simply due to it being released later and having them already bundled in the main game), will retool include the updates/dlcs from another language/region instead or exclude those?

**Comment by unexpectedpanda** [comment]
> maybe its just the clone list being less updated I guess compared to the main ones

There is no clone list for `Nintendo - Nintendo 3DS (Digital) (Updates and DLC)`. Contributions are welcome :)

> If a game does not has updates/dlc for the English/USA version (maybe simply due to it being released later and having them already bundled in the main game), will retool include the updates/dlcs from another language/region instead or exclude those?

It only operates within an individual DAT file, it doesn't have context on what exists in other DATs. So in this instance it'll take whatever the highest priority title is in the `(Updates and DLC)` DAT file, regardless of what the base title is in, say, `Nintendo - Nintendo 3DS (Digital) (CDN)`. This could mean that in your proposed scenario, yes, you could end up with updates that aren't linked to base titles. This is a No-Intro problem to solve, should they decide to extend the DAT format to encompass DLC/update linking to base titles.

**Comment by zealvix** [comment]
> > maybe its just the clone list being less updated I guess compared to the main ones
> 
> There is no clone list for `Nintendo - Nintendo 3DS (Digital) (Updates and DLC)`. Contributions are welcome :)
> 
> > If a game does not has updates/dlc for the English/USA version (maybe simply due to it being released later and having them already bundled in the main game), will retool include the updates/dlcs from another language/region instead or exclude those?
> 
> It only operates within an individual DAT file, it doesn't have context on what exists in other DATs. So in this instance it'll take whatever the highest priority title is in the `(Updates and DLC)` DAT file, regardless of what the base title is in, say, `Nintendo - Nintendo 3DS (Digital) (CDN)`. This could mean that in your proposed scenario, yes, you could end up with updates that aren't linked to base titles. This is a No-Intro problem to solve, should they decide to extend the DAT format to encompass DLC/update linking to base titles.

I was actually referring to "Unofficial - Nintendo - Nintendo 3DS (Digital) (Updates and DLC)" (can pick encrypted or decrypted when downloading the DAT) , which I can see clones being removed based on retool stats,  not " `Nintendo - Nintendo 3DS (Digital) (Updates and DLC)`" for the above, but the same applies I guess.

Thanks.
